dotnet/src/Microsoft.Agents.AI.Workflows/ChatProtocolExecutor.cs

```diff
@@ -8,23 +8,40 @@
 
 namespace Microsoft.Agents.AI.Workflows;
 
-internal class ChatProtocolExecutorOptions
+/// <summary>
+/// Provides configuration options for <see cref="ChatProtocolExecutor"/>.
+/// </summary>
+public class ChatProtocolExecutorOptions
 {
+    /// <summary>
+    /// Gets or sets the chat role to use when converting string messages to <see cref="ChatMessage"/> instances.
+    /// If set, the executor will accept string messages and convert them to chat messages with this role.
+    /// </summary>
     public ChatRole? StringMessageChatRole { get; set; }
 }
```

Comment:
https://github.com/microsoft/agent-framework/blob/main/dotnet/src/Microsoft.Agents.AI.Workflows/Specialized/AgentRunStreamingExecutor.cs#L13 and https://github.com/microsoft/agent-framework/blob/main/dotnet/src/Microsoft.Agents.AI.Workflows/Specialized/AIAgentHostExecutor.cs have been getting closer and closer to each other, and it'd be nice not to have both. I wonder if we could add as options here the things AgentRunStreamingExecutor still provides (namely role change) and then delete it. I also opened https://github.com/microsoft/agent-framework/issues/1599 to explore other more cross-cutting ways to do it.